drivers/soundwire/intel.c

```diff
@@ -412,9 +414,27 @@ static int intel_shim_init(struct sdw_intel *sdw, bool clock_stop)
 	usleep_range(10, 15);
 
 	if (!clock_stop) {
+		struct sdw_bus *bus = &sdw->cdns.bus;
+		struct sdw_master_prop *prop = &bus->prop;
+		u32 mclk = prop->mclk_freq;
+		u32 syncprd;
+
+		/*
+		 * The hardware relies on an internal counter,
+		 * typically 4kHz, to generate the SoundWire SSP -
+		 * which defines a 'safe' synchronization point
+		 * between commands and audio transport and allows for
+		 * multi link synchronization. The SYNCPRD value is
+		 * only dependent on the oscillator clock.
+		 */
+		if (mclk % 6000000)
```

Comment:
@plbossart I think it's easy enough and more transparent to just use `prop->mclk_freq` here directly. Also this check - is it only unclear to me? Is this something from the SDW protocol?

Reply:
I removed the variable as suggested, and flipped the values (wrong test).
Also added a comment, please re-check.